### PR TITLE
Added bulk download and import/export extensions

### PR DESCRIPTION
**Pull request by sanmadjack**
Added ob_flush to prevent a potential out-of-memory error
Changed chunk size to 1mb, makes a big difference for download speed

### CODE REVIEW

**Comment by sanmadjack** [comment]
Way too many changes upstream, I need to get my stuff back in sync

**Comment by shish** [comment]
Also yeah, external contributions seemed quiet, so I took the chance to refactor a bunch of things that had been bothering me for years (Including getting all of the unit tests to pass again!)

File: core/polyfills.php [hunk 1]
```diff
@@ -156,12 +156,13 @@ function stream_file(string $file, int $start, int $end): void
     try {
         set_time_limit(0);
         fseek($fp, $start);
-        $buffer = 1024 * 64;
+        $buffer = 1024 * 1024;
```
[the remark this answers is not in the document]
Comment (sanmadjack) [reply core/polyfills.php]: K, glad something good came out of my mess :)

File: ext/bulk_actions/main.php [hunk 2]
```diff
@@ -39,6 +39,11 @@ class BulkActionEvent extends Event
     public $action;
     /** @var array  */
     public $items;
+    /** @var PageRequestEvent  */
+    public $page_request;
```
[the remark this answers is not in the document]
Comment (sanmadjack) [reply ext/bulk_actions/main.php]: It's for something else I was working on that pulled the information about the request. I'll re-evaluate how I'm delivering that data so we can keep cli compatibility.

**Comment by sanmadjack** [comment]
Yeah, I noticed when I tried to rebase :D Always glad for a good re-factor, tech debt's better paid up-front. 

For my own sanity I'd like to start doing formatting passes before I check stuff in, so that your formatting passes don't cause me endless merge conflicts when re-basing. Would you mind sharing how you do your formatting pass?

My custom branch diverged a long time ago so I'm manually doing diffs against master so I can figure out what I've even still got to make pull requests for and sort them into logical commits. Getting the formatting in sync would simplify that.

**Comment by shish** [comment]
Formatting is done with php-cs-fixer - https://github.com/FriendsOfPHP/PHP-CS-Fixer - once that's installed, running "php-cs-fixer fix" in the project folder should format everything according to PHP standards (which aren't my favourite standards, but it beats every file being formatted differently :P)

**Pull request closed without merging**